File: ios/Classes/BannerView/BannerEventHandler.swift
//
//  BannerEventHandler.swift
//  clever_ads_solutions
//
//  Copyright © 2024 CleverAdsSolutions LTD, CAS.AI. All rights reserved.
//

import CleverAdsSolutions
import Flutter
import Foundation

private let channelName = "com.cleveradssolutions.plugin.flutter/banner."

class BannerEventHandler: EventHandler, CASBannerDelegate {
    init(_ flutterId: String) {
        super.init(channelName: channelName + flutterId)
    }

    func bannerAdViewDidLoad(_ view: CASBannerView) {
        success(["event": "onAdViewLoaded"])
    }

    func bannerAdView(_ adView: CASBannerView, didFailWith error: CASError) {
        success([
            "event": "onAdViewFailed",
            "data": error.message,
        ])
    }

    func bannerAdView(_ adView: CASBannerView, willPresent impression: CASImpression) {
        success([
            "event": "onAdViewPresented",
            "data": impression.toDict(),
        ])
    }

    func bannerAdViewDidRecordClick(_ adView: CASBannerView) {
        success(["event": "onAdViewClicked"])
    }
}

File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/bannerview/BannerEventHandler.kt
package com.cleveradssolutions.plugin.flutter.bannerview

import com.cleveradssolutions.plugin.flutter.bridge.base.EventHandler
import com.cleveradssolutions.plugin.flutter.util.toMap
import com.cleversolutions.ads.AdError
import com.cleversolutions.ads.AdImpression
import com.cleversolutions.ads.AdViewListener
import com.cleversolutions.ads.android.CASBannerView

private const val CHANNEL_NAME = "com.cleveradssolutions.plugin.flutter/banner."

class BannerEventHandler(
    flutterId: String
) : AdViewListener, EventHandler(CHANNEL_NAME + flutterId) {

    override fun onAdViewLoaded(view: CASBannerView) {
        success("event" to "onAdViewLoaded")
    }

    override fun onAdViewFailed(view: CASBannerView, error: AdError) {
        success(
            "event" to "onAdViewFailed",
            "data" to error.message
        )
    }

    override fun onAdViewPresented(view: CASBannerView, info: AdImpression) {
        success(
            "event" to "onAdViewPresented",
            "data" to info.toMap()
        )
    }

    override fun onAdViewClicked(view: CASBannerView) {
        success("event" to "onAdViewClicked")
    }

}